### PR TITLE
Multithread app by foxtra

### PR DESCRIPTION
**Pull request by Foxtra**
Hello,

there is my Pull Request for homework 8.
It is not finished yet, only tests were implemented.

Best Regards,
Sergei

### CODE REVIEW

**Comment by Foxtra** [comment]
Task link: #281

File: HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Services/CallParsingFromPreviousIterationService.cs [hunk 1]
```diff
@@ -0,0 +1,38 @@
+﻿using SiteParser.Core.Repositories;
+using SiteParser.Core;
+using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SiteParser.Core.Services
+{
+    public class CallParsingFromPreviousIterationService
```
Comment (chesdenis) [inline HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Services/CallParsingFromPreviousIterationService.cs]: incorrect and long name. Should be like UrlCollectorService

File: HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Services/DownloadPageService.cs [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+        {
+            this.downloader = downloader;
+        }
+
+        public string DownLoadPage(string requestUrl)
+        {
+            var downloadedResult = downloader.Download(requestUrl);
+            var resultPath = downloader.SaveIntoFile(downloadedResult);
+            return resultPath;
+            //using (HttpClient client = new HttpClient())
```
Comment (chesdenis) [inline HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Services/DownloadPageService.cs]: commented lines

**Comment by chesdenis** [comment]
> After some time I've launched my program an Exeption System.IO.DirectoryNotFoundException: 'Could not find a part of the path' in `DownloaderRepository` occurs.
> 
> изображение
> 
> But the directory exists
> 
> изображение

can you please copy path from debugging and paste into notepad, and then copy paste path from explorer and place on second line to compare?. It can be a long path issue, but need to double check this.

File: HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Repositories/ICleaner.cs [hunk 3]
```diff
@@ -0,0 +1,11 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Text;
+
+namespace SiteParser.Core.Repositories
+{
+    public interface ICleaner
```
Comment (chesdenis) [inline HW8/Sergey_Tkachev/SiteParser/SiteParser.Core/Repositories/ICleaner.cs]: bad name i think, cleaner of what?

**Pull request merged**